src/main/java/Dash.java

```diff
@@ -0,0 +1,305 @@
+                        if (stringArray[1].equals("X")) {
+                            e.markTaskAsDone();
+                        }
+                        list.add(e);
+                    } else {
+                        throw new CorruptedFileException("File is corrupted.");
+                    }
+                }
+                System.out.println("Existing data file found. Data has been loaded.");
+            } else {
+                if (f.getParentFile().exists()) {
+                    f.createNewFile();
+                } else {
+                    f.getParentFile().mkdir();
+                    f.createNewFile();
+                }
+                System.out.println("No existing data file found. New data file \"dash.txt\" has been created.");
+            }
+        } catch (IOException e) {
+            System.out.println("I/O error has occurred when creating new file: " + e.getMessage());
+        } catch (CorruptedFileException e) {
+            System.out.println("Please erase data in file: " + e.getMessage());
+        } finally {
+            System.out.println(HORIZONTAL_LINE);
+        }
+
+        System.out.println("Hello! I'm\n" + LOGO + "\nWhat can I do for you?");
+        System.out.println(HORIZONTAL_LINE);
+
+        while (true) {
+            String input = scanner.nextLine();
+            try {
+                if (input.equals("bye")) {
+                    break;
+                } else if (input.equals("list")) { //list command
+                    System.out.println(HORIZONTAL_LINE);
+                    int length = list.size();
+                    for (int i = 0; i < length; i++) {
+                        Task t = list.get(i);
+                        int num = i + 1;
+                        System.out.println(num + "." + t);
+                    }
+                    System.out.println(HORIZONTAL_LINE);
+                } else if (input.startsWith("mark")) { //mark command
+                    String[] string = input.split(" ", 2);
+                    int markNum = Integer.parseInt(string[1]);
+                    int num = list.size();
+                    if (markNum > num || markNum < 1) {
+                        System.out.println(HORIZONTAL_LINE);
+                        throw new WrongIndexException("Index is out of bounds.");
+                    } else {
+                        Task t = list.get(markNum - 1);
+                        t.markTaskAsDone();
+                        writeToFile(f.getPath(), list.get(0).toFile() + System.lineSeparator());
+                        for (int i = 1; i < num; i++) {
+                            appendToFile(f.getPath(), list.get(i).toFile() + System.lineSeparator());
+                        }
+                        System.out.println(HORIZONTAL_LINE);
+                        System.out.println("Nice! I've marked this task as done:");
+                        System.out.println(t);
+                        System.out.println(HORIZONTAL_LINE);
```

Comment:
Repeated code of printing, maybe can use a UI class to handle all of the printing